packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -93,8 +94,16 @@ export class OdspSummaryUploadManager {
                 this.forceAccessTokenViaAuthorizationHeader,
             );
             headers["Content-Type"] = "application/json";
+            const ifMatchHeader: string[] = [];
+            const relayServiceSessionId = this.relayServiceSessionId();
+            if (relayServiceSessionId !== undefined) {
```

Comment:
We should never allow undefined here. runtime already checks for not losing connection, but if there is a race condition, we should allow going with old ID (from old connection). Given that we have no flow (today) where we summarize without ever connecting to PUSH, we should simply disallow undefined type here, and assert in service class that upload always happens with valid ID